### PR TITLE
split logic into models with hypothesis data access

### PR DESCRIPTION
**Pull request by nathanstitt**

### CODE REVIEW

File: tutor/src/models/annotations/ux.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import {
+  computed, observable, action,
+} from 'mobx';
+
+export default class AnnotatorUX {
+
+  @observable isSummaryVisible = false;
+
+  @action.bound toggleSummary() {
+    console.log("TOGGLE SUM")
```
Comment (RoyEJohnson) [inline tutor/src/models/annotations/ux.js]: Do you want this left in?

File: tutor/src/components/annotations/annotation.jsx [hunk 2]
```diff
@@ -347,64 +338,52 @@ export default class AnnotationWidget extends React.Component {
...
-        // object as what is in the observable array
-        const newEntry = observable(dbEntryToPreload(response));
-
-        this.allAnnotationsForThisBook.push(newEntry);
-        return newEntry;
-      }));
+    return User.annotations.create({
+      documentId: this.props.documentId,
+      selection: this.savedSelection,
+      ecosystemId: this.props.ecosystemId,
```
Comment (RoyEJohnson) [inline tutor/src/components/annotations/annotation.jsx]: This is my fault, but ecosystemId in the annotation storage is actually courseId. That's necessary for the "open book in new tab" feature. I didn't get the variable names changed over.

File: tutor/src/models/annotations/ux.js (hunk 1 above)
Comment (nathanstitt) [reply tutor/src/models/annotations/ux.js]: Nope, that was a debug thing that got left in 😦   I removed in #1978

File: tutor/src/components/annotations/annotation.jsx (hunk 2 above)
Comment (nathanstitt) [reply tutor/src/components/annotations/annotation.jsx]: Ah ok. I'll update in the followup that adds the toggle

**Pull request merged**